### PR TITLE
fix: routing behind mediator 

### PR DESCRIPTION
**Pull request by dbluhm**
This is a reopen of PR #2535 (after I botched my git command...). Thanks again @codespree for the fix.

### CODE REVIEW

**Comment by dbluhm** [comment]
I'll take a quick test case onto this PR as a regression test.

**Comment by swcurran** [comment]
@dbluhm — can you cherry pick this into a v0.10.4 branch (or v0.10.3 and I’ll just tag it from there) and I’ll publish an image for testing.  Once we have that we can do a 0.10.4.

Thanks @codespree for the fix. Hoping that once I publish an image, you can give it a try.

**Comment by dbluhm** [comment]
@swcurran how about a 0.10.x branch and then we can delete the old 0.10.2 and 3 branches?

**Comment by swcurran** [comment]
Sure. As long as it is not 0.10.4.  That you mean the x literally. If you do mean 4, then add a ”v” in front.

**Comment by dbluhm** [comment]
https://github.com/hyperledger/aries-cloudagent-python/pull/2537 PR to 0.10.x. I'll let the tests run (I don't expect anything to be different enough between main and 0.10.x to cause issues but just in case) and merge once they complete.

**Comment by dbluhm** [comment]
https://github.com/hyperledger/aries-cloudagent-python/tree/0.10.x is now ready for an image publish

File: aries_cloudagent/resolver/default/tests/test_legacy_peer.py
Comment (codespree) [inline aries_cloudagent/resolver/default/tests/test_legacy_peer.py:1]: The test case accepts `did:key:z6Mko2LnynhGbkPQdZ3PQBUgCmrzdH9aJe7HTs4LKontx8Ge` and `did:key:z6Mko2LnynhGbkPQdZ3PQBUgCmrzdH9aJe7HTs4LKontx8Ge#z6Mko2LnynhGbkPQdZ3PQBUgCmrzdH9aJe7HTs4LKontx8Ge` but what if a legacy mediator sent a base58 key (without the `did:key:` prefix. Shouldn't that be accepted as well? @dbluhm @swcurran

**Comment by codespree** [comment]
> https://github.com/hyperledger/aries-cloudagent-python/tree/0.10.x is now ready for an image publish

Looks good! I also verified with a base58 key `F7cEyTgzUbFwHsTwC2cK2Zy8bdraeoMY8921gyDmefwK` by adding it as a `routingKey` and the code works well.

In the test case we already have `did:key:z6Mko2LnynhGbkPQdZ3PQBUgCmrzdH9aJe7HTs4LKontx8Ge`, `did:key:z6Mko2LnynhGbkPQdZ3PQBUgCmrzdH9aJe7HTs4LKontx8Ge#z6Mko2LnynhGbkPQdZ3PQBUgCmrzdH9aJe7HTs4LKontx8Ge`, perhaps we should add another service with the `routingKey` as `F7cEyTgzUbFwHsTwC2cK2Zy8bdraeoMY8921gyDmefwK`. That would cover all cases.

**Comment by codespree** [comment]
@dbluhm @swcurran I updated my local environments (mediator and web agent) to use the branch created:

```
pip install git+https://github.com/hyperledger/aries-cloudagent-python.git@0.10.x
```

I am happy to report that everything works as expected.

**Comment by swcurran** [comment]
Should I go ahead with the release?  What about the other question about the mediator - #2531?

**Comment by codespree** [comment]
> Should I go ahead with the release? What about the other question about the mediator - #2531?

@swcurran I have tested the mediator from a mobile agent and have also tested message queuing so, from my perspective, a release would be good.

Thanks @dbluhm for providing inputs on debugging and for merging the PR.

**Comment by dbluhm** [comment]
@swcurran I think this is good for a merge, based on @codespree's testing.

**Pull request merged**